examples/pagination-ts/src/types/react-table-local.d.ts

```diff
@@ -0,0 +1,6 @@
+declare module 'react-table' {
```

Comment:
This is probably the most useful file in the whole example.

That said, I really don't like this approach, it has a number of problems, some are rather unforgiving.

1. It's global.  So you can't configure two tables separately and then have two separate types for each table.
2. It's not at all obvious, requiring reading up on declaration-merging isn't a reasonable expectation to foist on users.

One alternative would be to just blindly add all of the plugins to an uber interface for the Table, it would work and give the user types, but then you lose the ability to validate that you are using options that match the configuration that you have chosen.  I've been avoiding this option because it feels like the wrong solution.  I don't want to stop at just having useful intellisense, I'd like to be able to enforce interface correctness.

Now if there was a way to derive these dynamic types from the plugin list...